File: strategy/java/PaypalPayment.java
package Strategy;

// UMl diyagramındaki ConcreteStrategy sınıfına denk gelmektedir.
// UML diyagramındaki Strategy arayüzüne denk gelen arayüzü uyguluyor.
class PaypalPayment implements IPayment {
    private String email;
    private String password;

    public PaypalPayment(String email, String password) {
        this.email = email;
        this.password = password;
    }

    public void pay(int amount) {
        // Kredi kartı ödeme işlemlerinin gerçekleştiği yer.
        System.out.println("Paid by Paypal.");
    }
}

File: strategy/typescript/paypal-payment.ts
import {IPayment} from "./payment.interface";

// UMl diyagramındaki ConcreteStrategy sınıfına denk gelmektedir.
// UML diyagramındaki Strategy arayüzüne denk gelen arayüzü uyguluyor.
export class PaypalPayment implements IPayment {
  constructor(
    private readonly email: string,
    private readonly password: string,
  ) {}

  public pay(amount: number): void {
    // Kredi kartı ödeme işlemlerinin gerçekleştiği yer.
    console.log('Paid by Paypal.');
  }
}
